### PR TITLE
Deprecate SingleGrid.position_agent

### PR DESCRIPTION
**Pull request by Tortar**
As noted in #1509 `SingleGrid.position_agent` should be deprecated in favor of `place_agent` and `move_to_empty`

### CODE REVIEW

File: mesa/space.py [hunk 1]
```diff
@@ -504,6 +504,14 @@ def position_agent(
         If x or y are positive, they are used.
         Use 'swap_pos()' to swap agents positions.
         """
+        warn(
+            (
+                "`position_agent` is being deprecated in favor of "
+                "`place_agent` and `move_to_empty`."
```
Comment (rht) [inline mesa/space.py:510]: I think the previous warning message describing when to use `place_agent` and `move_to_empty` is better.

Comment (Tortar) [reply mesa/space.py]: new way :-)

**Comment by rht** [comment]
The `position_agent` in the Schelling example should be replaced with `place_agent`.

**Comment by rht** [comment]
There is a merge conflict.

**Comment by rht** [comment]
LGTM

**Pull request merged**